On-the-fly compilation

StrikerRUS opened the pull request:
Finally, on-the-fly compilation is ready. Thanks to Rie Johnson (author of RGF) for the help with Visual Studio solution file!

Please upload new version to PyPI.

Fixed #49.

fukatani commented:
Thank you very much for pull PR. It's a great hacker's job.
**Installation is more refined than XGBOOST!**

To be honest, It is beyond my review ability, especially on Windows builds.

And we have to concern about licensing. 
Since RGF is GPL 3, should this repository also be GPL 3?

At least, we should add comment for RGF License in readme.

I do not mind rgf_python going to be GPL 3.
It does not make sense by wrapper alone.

StrikerRUS commented:
Thank you :-).

You can review only **sklearn.py** (I improved there only `_is_executable_response` and did some necessary permutations of code pieces with the aim of using `_TEMP_PATH` earlier in the module); **Readme.rst** and **setup.py** itself (for codestyle or something else).

To be honest, I know nothing about difference in licenses and what we should do 😢 . I know only that they both are open source licenses 😄 .
Maybe really the same license?.. I don't know...

fukatani commented on setup.py:
```diff
@@ -1,9 +1,179 @@
+                    os.remove(file_path)
+                elif os.path.isdir(file_path):
+                    rmtree(file_path, ignore_errors=True)
+            except Excetpion:
```
except Exception

StrikerRUS replied:
Oops.

fukatani commented:
There is no big difference between MIT License, BSD 3 clause, and Apache 2.0 License we are using now. However, please be careful when copying source code of copyleft type licenses like *GPL*.
When you copy the source code from GPL3 software, copy destination must also be GPL3 when you want to redistribute it.
When redistributing the code you copied from rgf_python, regardless some or all of code, you must make that software open source.

This limitation often called *GPL infection*, it is hated by companies.
Since you can not return it to GPL once, it is recommended to look for a commentary site in your native language.

fukatani commented:
If you want to switch to GPL3, replace the license file with that of GPL3 (copy include/rgf/COPYING to top and delete current license). Then add the following wording to at the end of the Readme.

**License**
rgf_python is distributed under the GNU public license version 3. Please read 
the file COPYING.
rgf_python includes RGF version 2.0 which is distributed under the GNU public license version 3.
original RGF distributed in http://tongzhang-ml.org/software/rgf .
Thank you to Rie Johnson and Tong Zhang.

StrikerRUS commented:
Am I right that rgf_python by itself let say has no value and it's no matter under what license it is distributed? Anyway users should download original RGF which is under GPLv3 and main restrictions come from there, right? So, we change license and it mean nothing to users.

> rgf_python includes RGF version 2.0 which is distributed ...

Why 2.0? 1.2 if I'm not mistaken...

fukatani commented:
> Am I right that rgf_python by itself let say has no value and it's no matter under what license it is distributed? Anyway users should download original RGF which is under GPLv3 and main restrictions come from there, right? So, we change license and it mean nothing to users.

Yes, you're right. Users will receive no effect.

For developer, when copying rgf_python to make another ML wrapper in other repository, it must also be GPL 3. However, I have no problem.

> Why 2.0? 1.2 if I'm not mistaken...

Yes, 1.2 is right.

StrikerRUS commented:
OK, I understood. I'll create commit with changing license.

StrikerRUS commented:
I replaced LICENSE file and updated references to it.
Please don't upload new version to PyPI and increment the version after merging this PR (do it after next one). I want to add last remaining parameter **memory_policy**. I've just received explanations of this param from Rie Johnson.

fukatani commented:
LGTM!
Thank you for your big contribution!
